### PR TITLE
return a new list in moniker range parser

### PR DESCRIPTION
**Pull request by 928PJY**
#3683 
 
return a new list in moniker range parser to avoid the cache be changed outside the parser.

### CODE REVIEW

File: src/docfx/lib/moniker/MonikerRangeParser.cs [hunk 1]
```diff
@@ -19,23 +19,28 @@ public MonikerRangeParser(MonikerDefinitionModel monikerDefinition)
         }
 
         public List<string> Parse(string rangeString)
```
Comment (superyyrrzz) [inline src/docfx/lib/moniker/MonikerRangeParser.cs]: return `IEnumerable<string>`?

File: src/docfx/lib/moniker/MonikerRangeParser.cs [hunk 2]
```diff
@@ -10,18 +10,18 @@ namespace Microsoft.Docs.Build
 {
     internal class MonikerRangeParser
     {
-        private readonly ConcurrentDictionary<string, Lazy<List<string>>> _cache = new ConcurrentDictionary<string, Lazy<List<string>>>();
+        private readonly ConcurrentDictionary<string, Lazy<IEnumerable<string>>> _cache = new ConcurrentDictionary<string, Lazy<IEnumerable<string>>>();
```
Comment (superyyrrzz) [inline src/docfx/lib/moniker/MonikerRangeParser.cs]: This can remain List as a private field

Comment (928PJY) [reply src/docfx/lib/moniker/MonikerRangeParser.cs]: yes, but it don't have to be, right?

---
In reply to: [236547338](https://github.com/dotnet/docfx/pull/3684#discussion_r236547338) [](ancestors = 236547338)

Comment (live1206) [reply src/docfx/lib/moniker/MonikerRangeParser.cs]: +1, it is enough to keep `IEnumerable<string> Parse(string rangeString)`

Comment (OsmondJiang) [reply src/docfx/lib/moniker/MonikerRangeParser.cs]: IReadOnlyList would be better?

**Pull request merged**